lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1604,7 +1604,7 @@ public function get_collection_params() {
 			),
 		);
 		$params['status'] = array(
-			'default'           => 'publish',
+			'default'           => 'attachment' === $this->post_type ? 'inherit' : 'publish',
```

Comment:
If I do a request logged out for `/media` and there's a bunch of posts with `post_status=inherit` which are attached to private posts, what does the response look like? Does it just remove the attachments I can't view (thus I get 6 results rather than the default 10) or do I get an error, or somethign else?